examples/cli/cli_test.go

```diff
@@ -0,0 +1,48 @@
+package cli
+
+import (
+	"bytes"
+	_ "embed"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/tetratelabs/wazero/internal/testing/require"
+)
+
+//go:embed testdata/cli.wasm
+var cliWasm []byte
+
+func TestRun(t *testing.T) {
+	tests := []struct {
+		args   []string
+		stdOut string
+	}{
+		{
+			args:   []string{"3", "1"},
+			stdOut: "result: 4",
+		},
+		{
+			args:   []string{"-sub=true", "3", "1"},
+			stdOut: "result: 2",
+		},
+	}
+
+	wasmPath := filepath.Join(t.TempDir(), "cli.wasm")
+	require.NoError(t, os.WriteFile(wasmPath, cliWasm, 0755))
+
+	for _, tc := range tests {
+		tt := tc
+		t.Run(strings.Join(tt.args, " "), func(t *testing.T) {
+			goCmd := filepath.Join(runtime.GOROOT(), "bin", "go")
```

Comment:
Oops guess this strategy won't work for the docker tests. I can try disabling these in the docker tests or something else after a first review pass